packages/gel/src/baseConn.ts

```diff
@@ -537,12 +551,31 @@ export class BaseRawConnection {
       if (this.stateCodec === INVALID_CODEC || this.stateCodec === NULL_CODEC) {
         wb.writeInt32(0);
       } else {
-        if (!this.stateCache.has(state)) {
-          const buf = new WriteBuffer();
-          this.stateCodec.encode(buf, state._serialise(), NOOP_CODEC_CONTEXT);
-          this.stateCache.set(state, buf.unwrap());
+        let encodedState: Uint8Array | null = null;
+
+        // Handle explicit transaction options isolation level, which should be
+        // set as the default isolation level for this query if it is not already
+        // in a transaction. If they match, do nothing.
+        if (
+          state.transactionOptions.isolation !==
+          state.config.get("default_transaction_isolation")
```

Comment:
I think this config value with typically be _unset_, so this path probably gets hit more often than it should considering that if you do nothing, the isolation will be `Serializable` which is also the default. But I didn't want to hard code the knowledge of the default into this driver code, so it seemed safer to just do an explicit comparison here and set the state more than needed.

Reply:
[Latest update to make transaction options fully optional](https://github.com/geldata/gel-js/pull/1270/commits/f0e62738231ec020f3c15f0b9320e2a152a023fd) makes this a bit more explicit: the isolation will be `undefined` (which will match the config value) if you do not set either, which will avoid this extra state copy/serialization.